### PR TITLE
Updated Sonos Cask to Version 8.0

### PR DESCRIPTION
**Pull request by marcoslater**
* Updated to use versioned URL.
* Added SHA verification.
* Added Zap for application folder.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **updating a cask**:

- [ ] `sha256` changed but `version` stayed the same ([what is this?][version-checksum]).  
      I’m providing public confirmation below.

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not already refused in [closed issues].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-drivers/pulls
[closed issues]: https://github.com/caskroom/homebrew-drivers/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/sonos.rb [hunk 1]
```diff
@@ -2,13 +2,13 @@
   version '8.0'
   sha256 '4362657796f74c66c1f342d50cc40233a04bc51a34bcc069bab57434c43541b7'
 
-  url 'https://update.sonos.com/software/mac/mdcr/SonosDesktopController80.dmg'
+  url "https://update.sonos.com/software/mac/mdcr/SonosDesktopController#{version.no_dots}.dmg"
   name 'Sonos'
-  homepage 'https://www.sonos.com/'
+  homepage 'http://www.sonos.com/'
```
Comment (commitay) [inline Casks/sonos.rb]: The homepage redirects to `http`

**Comment by marcoslater** [comment]
@commitay Thanks for correcting. Is there a list of all the available placeholders like "#{version.no_dots}" ? I didn't know that was a thing. ¯\_(ツ)_/¯

**Comment by commitay** [comment]
[`version`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/version.md)

**Comment by commitay** [comment]
Thanks for the PR!

**Pull request merged**